qcodes/utils/validators.py

```diff
@@ -488,32 +488,72 @@ def __repr__(self) -> str:
 
 class Arrays(Validator):
     """
-    Validator for numerical numpy arrays
+    Validator for numerical numpy arrays of numeric types (int, float, complex).
+    By default it validates int and float arrays.
+
+    Min and max validation is not supported for complex numbers.
 
     Args:
         min_value:  Min value allowed, default inf.
         max_value: Max value allowed, default inf.
         shape: The shape of the array, tuple of either ints or Callables taking
             no arguments that return the size along that dim as an int.
+        valid_types: Sequence of types that the validator should support. Should
+            be a subset of the supported types or None. If None all real
```

Comment:
Punctuation facilitates reading)
```suggestion
            be a subset of the supported types, or None. If None, all real
```